app/src/main/kotlin/com/simplemobiletools/clock/activities/MainActivity.kt

```diff
@@ -44,8 +44,8 @@ class MainActivity : SimpleActivity() {
         setupTabs()
         updateWidgets()
 
-        if (getNextAlarm().isEmpty()) {
-            ensureBackgroundThread {
+        lifecycleScope.launch(Dispatchers.IO) {
```

Comment:
do not use these lifecycleScope please, it cannot be read

Reply:
What do you want me to use in that case?

Reply:
Do you want to create an extension function around this or something?

Reply:
what does it do at all?

Reply:
`lifecycleScope.launch` launches a coroutine scope. Here I'm specifying that it be run on the IO thread pool because an existing function `rescheduleEnabledAlarms()` requires it to be run on the background thread.
The main reason for using coroutines is because `getEnabledAlarms` is a suspend function that needs to be called from a coroutine scope. 

What are your reservations using coroutines? It makes developer's lives so much easier

Using coroutines has many benefits in my opinion
- Is much more simpler and intuitive than using 
- Managing threads or executors & is also much more lightweight than creating a thread.
- Does not block the thread it is called on
- Exception handling is very straightforward
- Android lifecycle integrations like `lifecycleScope` that handle automatically canceling jobs based on the app's lifecycle state

If you'd still like to not use coroutine, I'll make the change

Reply:
yeah, this is a ridiculously overcomplicated solution for an easy problem... Lets remove all of it.

Reply:
Noted